### ISSUE TITLE
Element Preview - Details list layout

### ISSUE THREAD

**Issue by bezua** [microsoft-search/pnp-modern-search#175]
**Version used**
3.11.1

**Describe the bug**

Just upgraded from 3.7 to 3.11.1
I use 'Details list view' to present search results.

I used to use the following snippet to get element preview:

`<img class="document-preview-item img-preview" width="120" src="{{ServerRedirectedPreviewURL}}" data-src="{{ServerRedirectedPreviewURL}}" data-url="{{ServerRedirectedEmbedURL}}" />`

or 

`<img class="document-preview-item img-preview" width="120" src="{{ServerRedirectedPreviewURL}}" data-src="{{ServerRedirectedPreviewURL}}" data-url="{{Path}}" />`

After upgrading to 3.11.1 preview windows is gone. I can see thumbnail of a document, however preview window does not appear.

Note - preview window appears when switching to "simple list" layout

**To Reproduce**
1. Get two out of the box webparts:
- Search Box
- Search Results
and connect Search Results to Search Box
2. Configure Search Results web part to present data as "Details list".
3. Configure Search Results web part -> Manage Columns
 add two new columns using handlebars expr
(Preview 1) 
`<img class="document-preview-item img-preview" width="120" src="{{ServerRedirectedPreviewURL}}" data-src="{{ServerRedirectedPreviewURL}}" data-url="{{ServerRedirectedEmbedURL}}" />`

(Preview 2)
`<img class="document-preview-item img-preview" width="120" src="{{ServerRedirectedPreviewURL}}" data-src="{{ServerRedirectedPreviewURL}}" data-url="{{Path}}" />`

4. Save web part configuration and republish page.
5. Preview of a document does not work
**Expected behavior**
I expected preview window to be available

**Screenshots**

**Desktop (please complete the following information):**
 - Browser: confirmed on chrome and edge

**Additional context**

**Comment by bezua** [comment]
Trying to understand what's going on.
For now
BaseTemplateService._initDocumentPreviews
executes when there are no results generated in the DOM yet. Which means no targets where PreviewContainer could be attached to.

**Comment by wobba** [comment]
Hi,
When you say preview, do you mean a thumbnail inline? The Simple list, tiles and carousel has this, but not the details list.

That said, I see thumbs are missing for image files in the simple list, which is a bug.

**Comment by bezua** [comment]
Hi,
By preview I understand the ability to preview a document in a dedicated iframe with scrolls. It appears after clicking/hover on thumbnail image.
The code is scanning DOM (with DomHelper) to attach appropriate event (selecting html elements by class name), however for some reason DOM is not fully generated yet when componentDidUpdate is called. 
I don’t understand it, because according to react documentation at this moment DOM shall be fully generated. 
My trick with setTimeout is working well in my fork, however I did not decide to make a Pull Request because I still don’t understand what exactly is going on.

**Comment by wobba** [comment]
Got it now and will see of I can repro.

**Comment by FranckyC** [comment]
Hi @bezua I think I understand what is going on. Unfortunately I can't reproduce this issue but you can try something to validate if I'm right. Try to add the following method in the `DetailsListcomponent.tsx` file and `DetailsListComponent` class:

```
    public componentDidUpdate() {
        TemplateService.initPreviewElements();
    }
```
The details list use its own Handlebars template compilation process. Because it is a web component, it does not follow the main Web Part template processing so probably the preview init is already done when the web component finishes its stuff.

**Comment by FranckyC** [comment]
Ok got it. Should be fixed by #213. You can test and tell me if it works for you.

**Comment by bezua** [comment]
It works, thank you
